Allow users to clear locally stored donation history

cesonha opened the pull request:
### What does this do?
This will add a new preference in the SettingsFragment that enables the user to delete the local donation history. All this done respecting the business rules defined in the Phabricator ticket. The new preference only shows if the donation history is not empty and after the user deletes it the preference hides.

### Why is this needed?
Please refer to the Phabricator's link Background section
**Phabricator:**
https://phabricator.wikimedia.org/T380662

### Screenshot
![image](https://github.com/user-attachments/assets/473d859a-416b-46c4-8291-e7e7f3a31f17)
![image](https://github.com/user-attachments/assets/09a9d7f3-9d2b-4d3c-9fcf-f6b3cacb0c3d)

cesonha commented:
Hi everyone, this is my first contribution and I have some doubts that I would be happy to get some clarification:

- I saw that the iOS and Android Settings screen have a different UI and the preferences are organized in different sections, how can I know that the way I did is approved by the UX/UI team?
- What should I do regarding the translation/localization, do I need to do it by myself?
- What about unit tests, I understand that some classes might be easier to test than others, is there a discussion for that? I want to add tests to my PR but some guidance would be nice!

cooltey commented:
Hi @cesonha, thanks for submitting the PR!

> Hi everyone, this is my first contribution and I have some doubts that I would be happy to get some clarification:
> 
>     * I saw that the iOS and Android Settings screen have a different UI and the preferences are organized in different sections, how can I know that the way I did is approved by the UX/UI team?

If you have claimed the ticket, you can tag [HNordeenWMF](https://phabricator.wikimedia.org/p/HNordeenWMF/) or [SChekfa-WMF](https://phabricator.wikimedia.org/p/SChekfa-WMF/) by attaching the screenshot you have in the comment.

>     * What should I do regarding the translation/localization, do I need to do it by myself?

The way we add a translation, we will add the same `<string name="...">` you have added to the default `strings.xml` to the `qq/strings.xml`, with a description of the string item.

>     * What about unit tests, I understand that some classes might be easier to test than others, is there a discussion for that? I want to add tests to my PR but some guidance would be nice!
> 

I do not think this PR would need a test case.

>     * Do we need a summary for this preference?

You can put this comment in the Phabacator ticket and tag our PM ( the one who created the ticket).

cesonha commented:
Thanks @cooltey , just did as you've said and commented in phabricator tagging them!

cesonha commented:
Hi @cooltey, it seems to be approved by the design and product team :) I've rebased with the main branch and added the string descriptions as you've said we should, should I just wait for code review/testing? :) TY

cooltey commented on app/src/main/res/values-qq/strings.xml:
```diff
@@ -1706,4 +1706,10 @@
   <string name="survey_dialog_submit">Positive action button text for the survey dialog</string>
   <string name="survey_dialog_input_hint">Hint label for the survey dialog input field.</string>
   <string name="survey_dialog_submitted_snackbar">Text for the snackbar message the feedback form in the survey dialog that indicates the feedback has been submitted.</string>
+  <string name="dialog_confirm_delete_donor_history_title">Title text shown at the top of the confirmation dialog when deleting donation history</string>
+  <string name="dialog_confirm_delete_donor_history_message">Explanatory message shown in the confirmation dialog that clarifies the scope of deletion - that it only affects donations recorded on the current device</string>
+  <string name="dialog_confirm_delete_donor_history_delete">Label for the confirmation button in the deletion dialog that executes the deletion action</string>
+  <string name="dialog_confirm_delete_donor_history_cancel">Label for the button that dismisses the deletion dialog without deleting anything</string>
+  <string name="preference_title_delete_local_donation_history">Title text for the settings menu item that opens the donation history deletion dialog</string>
+  <string name="donor_history_deleted_message_snackbar">Success message shown briefly at the bottom of the screen after donation history has been deleted</string>
```
For the string items, I suggest using `donation_history` instead of `donor_history` to reduce confusion compared to the previous experimental feature.

cesonha commented on app/src/main/java/org/wikipedia/settings/SettingsPreferenceLoader.kt:
```diff
@@ -66,12 +66,40 @@ internal class SettingsPreferenceLoader(fragment: PreferenceFragmentCompat) : Ba
 
+    private val shouldShowDeleteLocalDonationHistoryPreference get() = Prefs.donationResults.isNotEmpty()
+
+    private fun deleteLocalDonationHistory() {
```
agreed!

cesonha commented on app/src/main/java/org/wikipedia/settings/SettingsPreferenceLoader.kt:
```diff
@@ -66,12 +66,40 @@ internal class SettingsPreferenceLoader(fragment: PreferenceFragmentCompat) : Ba
             (findPreference(R.string.preference_key_logout) as LogoutPreference).activity = activity
         }
+
+        if (shouldShowDeleteLocalDonationHistoryPreference) {
```
makes sense ty

cooltey reviewed (approved):
All looks good! Thanks for the updates.
